### ISSUE TITLE
Vendor method annotations not detected

### ISSUE THREAD

**Issue by ro0NL** [vimeo/psalm#10336]
I have this simple code:

```php
final class RedisCounter
{
    public function __construct(
        ClientInterface $redis,
    ) {
        /** @psalm-trace $redis */
        $redis->copy();
    }
}
```

I get 

```
INFO: Trace - src/Counter/RedisCounter.php:15:9 - $redis: Predis\ClientInterface (see https://psalm.dev/224)
        /** @psalm-trace $redis */
        $redis->copy();


ERROR: UndefinedMagicMethod - src/Counter/RedisCounter.php:15:17 - Magic method Predis\ClientInterface::copy does not exist (see https://psalm.dev/219)
        $redis->copy();
```

Upstream interface: https://github.com/predis/predis/blob/v2.2.2/src/ClientInterface.php

It looks like a psalm bug :thinking:

**Comment by ro0NL** [comment]
![image](https://github.com/vimeo/psalm/assets/1047696/8b844412-2e67-4890-a137-10b62a9f9319)

phpstorm is just fine

**Comment by robchett** [comment]
psalm couldn't parse the docblock for `Predis\ClientInterface` due to a syntax error
 
> `* @method array             cmsincrby(string $key, string|int...$itemIncrementDictionary)`

it should be 

> `* @method array             cmsincrby(string $key, string|int ...$itemIncrementDictionary)`

**Comment by ro0NL** [comment]
omfg. thanks :)
